### PR TITLE
fix+build(deps): bump `fedimint-core` jsonrpsee dependency,  and fix `SafeUrl` parsing

### PR DESCRIPTION
**Pull request by oleonardolima**

### CODE REVIEW

**Comment by elsirion** [comment]
Can you give some background on what the `jsonrpsee` bump gives us?

**Comment by oleonardolima** [comment]
> Can you add some background on what the `jsonrpsee` bump gives us?

At the current state (this PR), we are only getting the fix on default ports (nbd).

But the main idea was to get it to the latest while my changes on jsonrpsee repository haven't landed yet, also to make it easier to fix any broken API if it had any (It didn't)

**Comment by elsirion** [comment]
Needs rebase

**Comment by elsirion** [comment]
Please mark as ready for review once it is :)

**Comment by oleonardolima** [comment]
> Please mark as ready for review once it is :)

I've rebased and fixed the conflicts, now it's ready to review

File: fedimint-core/src/api.rs [hunk 1]
```diff
@@ -726,13 +726,13 @@ impl JsonRpcClient for WsClient {
...
             .max_concurrent_requests(u16::MAX as usize)
-            .build(url_to_string_with_default_port(url)) // Hack for default ports, see fn docs
+            .build(url.as_str()) // Hack for default ports, see fn docs
             .await;
 
         #[cfg(target_family = "wasm")]
         WsClientBuilder::default()
             .max_concurrent_requests(u16::MAX as usize)
-            .build(url_to_string_with_default_port(url)) // Hack for default ports, see fn docs
+            .build(url.as_str()) // Hack for default ports, see fn docs
```
Comment (dpc) [inline fedimint-core/src/api.rs:735]: Stale comments now?

**Review by dpc — DISMISSED** [review]

File: fedimint-core/src/api.rs (hunk 1 above)
Comment (elsirion) [reply fedimint-core/src/api.rs:735]: Yes, should be removed

**Review by elsirion — COMMENTED** [review]
Looks good, but please remove the stale comments

File: fedimint-core/src/api.rs (hunk 1 above)
Comment (oleonardolima) [reply fedimint-core/src/api.rs]: done [ebed3b7](https://github.com/fedimint/fedimint/pull/3454/commits/ebed3b7e8ccb5d5c319fdf2455a5b5c0b41eec49)

**Review by elsirion — APPROVED** [review]

**Pull request merged**